src/services/event.js

```diff
@@ -16,6 +16,93 @@ function start ({ communication }) {
     key: 'event'
   })
 
+  responder.on('getHistory', async (req) => {
+    const platformId = req.platformId
+    const env = req.env
+    const { Event } = await getModels({ platformId, env })
+
+    const {
+      order,
+
+      page,
+      nbResultsPerPage,
+
+      groupBy,
+
+      id,
+      createdDate,
+      eventType: type,
+      objectType,
+      objectId,
+      emitter,
+      emitterId
+    } = req
+
+    const queryBuilder = Event.knex()
+
+    const retentionLimitDate = getRetentionLimitDate()
+
+    const paginationMeta = await performHistoryQuery({
+      queryBuilder,
+      schema: Event.defaultSchema,
+      table: Event.tableName,
+      groupBy,
+      retentionLimitDate,
+      filters: {
+        ids: {
+          dbField: 'id',
+          value: id,
+          transformValue: 'array',
+          query: 'inList'
+        },
+        createdDate: {
+          dbField: 'createdTimestamp',
+          value: createdDate,
+          query: 'range',
+          defaultValue: { gte: retentionLimitDate }
+        },
+        types: {
+          dbField: 'type',
+          value: type,
+          transformValue: 'array',
+          query: 'inList'
+        },
+        objectTypes: {
+          dbField: 'objectType',
+          value: objectType,
+          transformValue: 'array',
+          query: 'inList'
+        },
+        objectIds: {
+          dbField: 'objectId',
+          value: objectId,
+          transformValue: 'array',
+          query: 'inList'
+        },
+        emitter: {
+          dbField: 'emitter',
+          value: emitter
+        },
+        emitterIds: {
+          dbField: 'emitterId',
+          value: emitterId,
+          transformValue: 'array',
+          query: 'inList'
+        }
```

Comment:
How’s performance with fields not indexed (such as emitter[Id])?

Reply:
For event, only `emitter` isn't indexed because there are only 3 currently allowed values (core, task, custom).
The query planner won't choose to use the index on that column even if it exists.

Concerning the response time, the test platform with 3M on non-compressed rows (filter on columns are only allowed within the retention log period), it's pretty good (<100ms). That can be explained by the fact the different emitter values are evenly distributed.